### PR TITLE
feat: more meaningful error message for kustomize issues: #8218

### PR DESCRIPTION
**Pull request by palladius**
a kustomize error is now surfaced correctly pinpointing to the actual file yielding error.

Fixes: #nnn 
**Related**:  #8218
**Merge before/after**: nope.

**Description**
This fixes the #8218 bug by adding meaningful new error.

**User facing changes (remove if N/A)**

Just more meaningful error message

**Follow-up Work (remove if N/A)**

`make test` returns zero

### CODE REVIEW

**Comment by aaron-prindle** [comment]
Thanks for this @palladius!  I think the best options for this are:
1.  Just have the kustomize parse info added the the error message (as my comment suggests)
2.  Create a new actionable err (https://skaffold.dev/docs/references/api/grpc/#actionableerr) and emit that I'm this case

**Comment by palladius** [comment]
Aaron, happy to do (1). 

Have you checked the ouptut delta in https://github.com/GoogleContainerTools/skaffold/issues/8218 ?

Is there any other error output you believe is missing?

**Review by palladius — COMMENTED** [review]
test comment

File: pkg/skaffold/render/generate/util.go [hunk 1]
```diff
@@ -100,7 +100,7 @@ func DependenciesForKustomization(dir string) ([]string, error) {
 
 	content := kustomization{}
 	if err := yaml.Unmarshal(buf, &content); err != nil {
-		return nil, err
+		return nil, fmt.Errorf("🚨 [Error 1201] Kustomization parse error in %v: %w", path, err)
```
[the remark this answers is not in the document]
Comment (palladius) [reply pkg/skaffold/render/generate/util.go:103]: Thanks Aaron! I'll remove the first part and send for review.

**Comment by palladius** [comment]
I've fixed this in https://github.com/GoogleContainerTools/skaffold/commit/f79445e3d29dc74c8d0c54dd3bf2f9a0150aeb11

**Comment by palladius** [comment]
it seems to me the integration test failed for some timeout - I dont see a proper error. Any chance we can re-issue the IT?

**Comment by aaron-prindle** [comment]
+1 the PR looks good, I've re-triggered the integration test as I believe it was a flaky test.  Once we get all green we can merge

**Review by aaron-prindle — APPROVED** [review]
LGTM!  Thanks for this @palladius!!

**Pull request merged**